src/mediumComplexityProblem3/SwapNumbers.java

```diff
@@ -0,0 +1,52 @@
+package mediumComplexityProblem3;
+/*Q3. Given an array of ints, swap the first and last elements in the array. Return the modified array.
+If the last array value is prime no.
+swapEnds([1, 2, 3, 7]) → [7, 2, 3, 1]
+swapEnds([1, 2, 3]) → [3, 2, 1]
+swapEnds([1, 6, 7, 9, 8]) → [1, 6, 7, 9, 8] */
+
+import java.util.Arrays;
+
+public class SwapNumbers {
+    private int temp;
+    private int lastNum;
+    private boolean flag = false;
+
+    public void swapEnds(int[] arr) {
+        //To find last array value
+        lastNum = arr[arr.length - 1];
+        //To check whether the last array value is a prime number
+        for (int i = 2; i <= (lastNum / 2); ++i) {
+            if (lastNum % i == 0) {
+                flag = true;
+                System.out.println("Last Array value:" + lastNum + " is not a prime number");
+                break;
+            }
+        }
+        //Swap the first and last elements in the array,If the last array value is prime number
+        if (flag == false) {
```

Comment:
in if statement always check for true condition

Reply:
As per updated code 'flag' is not required so removed from code